R/comparePlot.R

```diff
@@ -100,31 +101,122 @@ comparePlot <- function(compareDF,
-                            msg = "Plot type must be either canvasXpress or ggplot.")
-    assertthat::assert_that(length(symbolSize) == 4,
-                            length(symbolShape) == 4,
-                            length(symbolColor) == 4,
-                            length(symbolFill) == 4,
-                            msg = paste0("All specified symbol arguments must be of length 4, ",
-                                         "including symbolSize, symbolShape, symbolColor, and symbolFill."))
+                           msg = "Need at least two numeric columns in compareDF.")
+    plotType <- tolower(plotType)
+    assertthat::assert_that(plotType %in% c("canvasxpress", "ggplot"),
+                                       msg = "Plot type must be either canvasXpress or ggplot.")
+    if (any(is.null(pThreshold),
+            !is.numeric(pThreshold),
+            length(pThreshold) != 1)) {
+        warning("pThreshold must be a singular value of class numeric. Assigning default value '0.01'.")
+        pThreshold <- 0.01
+    }
+
+    if (!is.null(title) &&
+        !all(is.character(title), length(title) == 1)) {
+        warning("title must be a singular value of class character. Assigning default value 'NULL'.")
+        title <- NULL
+    }
+
+    if (!is.null(xlab) &&
+        !all(is.character(xlab), length(xlab) == 1)) {
+        warning("xlab must be a singular value of class character. Assigning default value 'NULL'.")
+        xlab <- NULL
+    }
+
+    if (!is.null(ylab) &&
+        !all(is.character(ylab), length(ylab) == 1)) {
+        warning("ylab must be a singular value of class character. Assigning default value 'NULL'.")
+        ylab <- NULL
+    }
+
+    if (any(is.null(symbolSize),
+            !is.numeric(symbolSize),
+            length(symbolSize)  != 4)) {
+        warning("symbolSize must be a vector of 4 integer values. Assigning default values 10, 10, 10, 4.")
+        symbolSize  <-  c(10, 10, 10, 4)
+    }
+
+    if (any(is.null(symbolFill),
+            !is.character(symbolFill),
+            length(symbolFill)  != 4)) {
+        warning("symbolFill must be a vector of 4 character values. Assigning default values 'darkgoldenrod1', 'deepskyblue4', 'red3', 'grey25'.")
+        symbolFill <- c("darkgoldenrod1", "deepskyblue4", "red3", "grey25")
+    }
+
+    if (any(is.null(transparency),
+            !is.numeric(transparency),
+            length(transparency) != 1,
+            transparency <= 0,
+            transparency > 1)) {
+        warning("transparency must be a singular value of class numeric and must be between 0 and 1. Assigning default value '0.5'.")
+        transparency <- 0.5
+    }
+
+    if (!is.null(crosshair) &&
+        !all(is.character(crosshair), length(crosshair) == 1)) {
+        warning("crosshair must be a singular value of class character or 'NULL' to disable. Assigning default value 'grey50'.")
+        crosshair <- "grey50"
+    }
+
+    if (!is.null(referenceLine) &&
```

Comment:
I think you can give multiple reference lines in ggplot. Atleast that is what i found in plotPvalHist. Can you check? If that is the case, then you need to have separate checks for ggplot and cxplot